### ISSUE TITLE
Android release build error DexArchiveMergerException, duplicate classes

### ISSUE THREAD

**Issue by YeeWong11** [wix/Detox#2848]
### Describe the bug
Building the Android release build with the following command gives an error. Debug builds works without a problem.
```
"export BUNDLER_OVERRIDE_SUFFIX=e2e && export RCT_NO_LAUNCH_PACKAGER=true && cd android && ./gradlew assembleStagingRelease  assembleStagingReleaseAndroidTest -DtestBuildType=release && cd .."
```

```
* What went wrong:
Execution failed for task ':app:mergeExtDexStagingRelease'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
     Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
     Type androidx.coordinatorlayout.widget.CoordinatorLayout$1 is defined multiple times: /Users/devguy/.gradle/caches/transforms-2/files-2.1/5330eb3e932d450b979e7e466b7a21ef/coordinatorlayout-1.1.0-runtime/classes.dex, /Users/devguy/.gradle/caches/transforms-2/files-2.1/d518e896c4519c34d7d3f16d8ee4548c/coordinatorlayout-1.1.0-runtime/classes.dex
```

If we add an `&&` between the 2 archive commands, the build will work.
```
"export BUNDLER_OVERRIDE_SUFFIX=e2e && export RCT_NO_LAUNCH_PACKAGER=true && cd android && ./gradlew assembleStagingRelease &&  assembleStagingReleaseAndroidTest -DtestBuildType=release && cd .."
```

### Environment (please complete the following information):
Detox: 18.16.0
React Native: 0.64.2
Node: v14.15.4
Device: emulator Pixel_API_28_AOSP with test-butler
OS: Android
Test-runner (select one): jest-circus

**Comment by noomorph** [comment]
Any idea why this is exactly relevant to Detox? Something in logs?

**Comment by YeeWong11** [comment]
@noomorph Nothing directly pointing Detox, I just thought there could be some relationships between this error and part of the Detox Android setup where we added the Java class DetoxTest.

**Comment by noomorph** [comment]
@YeeWong11, and how did your addition look like? e.g...:

```diff
.../myfilename
-deleted
+added
```

Did it work prior to adding `DetoxTest`? I mean, `... ./gradlew assembleStagingRelease  assembleStagingReleaseAndroidTest ... `...

**Comment by Andarius** [comment]
@noomorph I'm having a similar issue after upgrading from `18.13.0` to `18.18.1` when running build in release mode for Android

`./gradlew clean && ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release`

```
* What went wrong:
Execution failed for task ':app:mergeExtDexRelease'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.DexMergingTaskDelegate
   > There was a failure while executing work items
      > A failure occurred while executing com.android.build.gradle.internal.tasks.DexMergingWorkAction
         > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
           Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
           Type androidx.appcompat.app.ActionBar$DisplayOptions is defined multiple times: /home/julien/.gradle/caches/transforms-2/files-2.1/9b66fac3130eaea21d606a287305777f/appcompat-1.2.0-runtime/classes.dex, /home/julien/.gradle/caches/transforms-2/files-2.1/b77ba274fe08f38c361bb8bc8dd75f52/appcompat-1.2.0-runtime/classes.dex
```

**Comment by Andarius** [comment]
I tried passing 

```
    androidTestImplementation('com.wix:detox:+') {
        exclude group: 'androidx.appcompat'
        exclude group: 'androidx.viewpager2'
        exclude group: 'androidx.fragment'
        exclude group: 'androidx.vectordrawable'
    }
 ```
but still get ` Type android.support.v4.app.INotificationSideChannel is defined multiple times`. 
Maybe someone more Android savy could see what's wrong here ?

**Comment by hanno-jonlay** [comment]
I can confirm that I had the same issue, with much the same environment configuration (including Test Butler), and observed the same solution.

I tried various debugging steps before I found this thread:

- My usual `android build` commands succeeded, and I only experienced the Dex merge failure when I ran my Detox build in release mode for testing - observed both locally and on CI.
- I tried upgrading all Detox/RN/Gradle versions to the latest - issue still observed.
- I tried commenting our Test Butler - no effect there - issue still observed.

@Andarius solution worked for me:

- I also observed that the issue didn't exist on `18.13.0`, but did after `18.14.0`
- I also had the `com.google.android.material` dependency in my project - not directly added to my `android/app/gradle.build`, but rather introduced via `react-native-screens` - https://github.com/software-mansion/react-native-screens/blob/master/android/build.gradle#L41
- The use of `exclude group: 'com.google.android.material'` allows me to build successfully with all dependencies at their current, latest versions: `react-native@0.64.2`, `detox@18.8.1` (and `react-native-screens@3.4.0`)
